qiime2/sdk/result.py

```diff
@@ -556,3 +469,132 @@ def get_index_paths(self, relative=True):
     def _repr_html_(self):
         from qiime2.jupyter import make_html
         return make_html(str(self._archiver.path))
+
+
+class ResultCollection:
```

Comment:
If you add `.type` as a property, then some interface logic can end up being simpler

Reply:
I would want the return from `.type` to be a `TypeExp` of `Collection[Whatever]` right? I'm not positive how to construct one of those. I can get the internal bit by just grabbing it off the first element in the collection, but I'm not sure about wrapping it in the `Collection` part.